databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,471 @@
+        return 'publisher.neo4j'
+
+    def _create_indices(self, node_file: str) -> None:
+        """
+        Go over the node file and try creating unique index
+        :param node_file:
+        :return:
+        """
+        LOGGER.info('Creating indices. (Existing indices will be ignored)')
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            for node_record in pandas.read_csv(node_csv,
+                                               na_filter=False).to_dict(orient='records'):
+                label = node_record[NODE_LABEL]
+                if label not in self._labels:
+                    with self._driver.session(database=self._db_name) as session:
+                        try:
+                            session.write_transaction(self._try_create_index, label)
+                        except Neo4jError as e:
+                            if 'An equivalent constraint already exists' not in e.__str__():
+                                raise
+                            # Else, swallow the exception, to make this function idempotent.
+                    self._labels.add(label)
+
+        LOGGER.info('Indices have been created.')
+
+    def _write_transactions(self,
+                            records: List[dict],
+                            stmt: str) -> None:
+        params_list = []
+        start_idx = 0
+        while start_idx < len(records):
+            stop_idx = min(start_idx + self._transaction_size, len(records))
+
+            for i in range(start_idx, stop_idx):
+                params_list.append(self._create_props_param(records[i]))
+
+            with self._driver.session(database=self._db_name) as session:
+                session.write_transaction(self._execute_statement, stmt, {'batch': params_list})
+
+            params_list.clear()
+            start_idx = start_idx + self._transaction_size
+
+    def _publish_node(self, node_file: str) -> None:
+        """
+        Iterate over the csv records of a file, each csv record transform to Merge statement
+        and will be executed.
+        All nodes should have a unique key, and this method will try to create unique index on
+        the LABEL when it sees first time within a job scope.
+        Example of Cypher query executed by this method:
+        MERGE (col_test_id1:Column {key: 'presto://gold.test_schema1/test_table1/test_id1'})
+        ON CREATE SET col_test_id1.name = 'test_id1',
+                      col_test_id1.order_pos = 2,
+                      col_test_id1.type = 'bigint'
+        ON MATCH SET col_test_id1.name = 'test_id1',
+                     col_test_id1.order_pos = 2,
+                     col_test_id1.type = 'bigint'
+
+        :param node_file:
+        :return:
+        """
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            all_node_records = pandas.read_csv(node_csv, na_filter=False).to_dict(orient="records")
+            if len(all_node_records) > 0:
```

Comment:
if this condition is false, then stmt will be undefined the the line below will fail. Maybe the case of empty files just doesn't need to be handled.